### DISCUSSION TITLE
Conflicting single_process_type and specialized_init_system checks

### DISCUSSION THREAD

**Discussion by sysarch-repo** [lfn-cnti/testsuite#2032] [General]
Is there any plan to address the conflict between the single_process_type and specialized_init_system checks?
For example the single_process_type could be made tolerant to dumb-init, tini and s6-overlay?

🎬 Testing: [single_process_type]
resource: {kind: "Deployment", name: "dns-dserver", namespace: "default"}, pod dns-dserver-b447b56ff-4nmtx and container: dserver has more than one process type (/sbin/tini--/usr/local/bin/dns-app, /usr/local/bin/dns-app)

As for s6 overlay and for everyone's benefit I have addressed the issue with s6 overlay not running on a read-only file system (https://github.com/just-containers/s6-overlay/issues/578)

**Comment by taylor** [comment]
Good idea. It appears https://github.com/cnti-testcatalog/testsuite/issues/2030 is related to this.  It will need to be addressed for all init systems.  Ideally the test will not be skipped but instead will tolerate the init-related process in addition to the normal process. Eg. 
/usr/local/bin/dumb-init--/usr/local/bin/dns-app should be allowed along with  /usr/local/bin/dns-app

**Comment by sysarch-repo** [comment]
@taylor I have created the issue https://github.com/cnti-testcatalog/testsuite/issues/2038 to address this.
